libvast/src/system/spawn_index.cpp

```diff
@@ -58,20 +57,8 @@ spawn_index(node_actor::stateful_pointer<node_state> self,
     VAST_VERBOSE("using customized indexing configuration {}", index_config);
   }
   if (opt("vast.use-legacy-query-scheduler", false)) {
-    auto handle = self->spawn(
-      legacy_index, accountant, filesystem, archive, catalog, type_registry,
-      indexdir,
-      // TODO: Pass these options as a vast::data object instead.
-      opt("vast.store-backend", std::string{sd::store_backend}),
-      opt("vast.max-partition-size", sd::max_partition_size),
-      opt("vast.active-partition-timeout", sd::active_partition_timeout),
-      opt("vast.max-resident-partitions", sd::max_in_mem_partitions),
-      opt("vast.max-taste-partitions", sd::taste_partitions),
-      opt("vast.max-queries", sd::num_query_supervisors),
-      std::filesystem::path{opt("vast.catalog-dir", indexdir.string())},
-      std::move(index_config));
-    VAST_VERBOSE("{} spawned the index", *self);
-    return caf::actor_cast<caf::actor>(handle);
+    VAST_WARN("The `vast.use-legacy-query-scheduler` is not supported anymore "
+              "and will be ignored.");
```

Comment:
I would like to have this in a dedicated PR.
And I would prefer to treat this as an error.

Reply:
I will split it out before the final review, but I disagree about making it an error because that would be a breaking change that would technically require a VAST 3.0 release.

Reply:
I don't agree with overly strict interpretations of versioning guidelines, but I don't care strongly enough about this particular case, so fine by me.

For the actual change though, can you try to revert 8beee922e94d50fd41384575ad925bc9a34e2eeb instead of the manual deletion? I'm afraid that we'll leave dead code lying around.

Reply:
Can we move this deprecation warning to the other deprecation warnings in `main()` in `vast/vast.cpp`?